### PR TITLE
Create new OMIS invoice when billing fields change

### PR DESCRIPTION
**Pull request by marcofucci**
This allow advisers to edit invoice/billing details when an order is in 'Quote Accepted' and creating a new invoice as a consequence.

The net cost (the one shown in the quote) cannot change after the quote is accepted but the VAT details can and therefore the gross cost on the invoice as well.

### CODE REVIEW

File: datahub/omis/order/serializers.py [hunk 1]
```diff
@@ -251,6 +263,22 @@ def create(self, validated_data):
...
+        """
+        Update invoice details if any of the invoice related fields has changed.
+        """
+        with transaction.atomic():
+            instance = super().update(instance, validated_data)
+
+            # update invoice details if necessary
+            if (
+                instance.status == OrderStatus.quote_accepted
+                and (set(ORDER_FIELDS_INVOICE_RELATED) & set(validated_data))
```
Comment (reupen) [inline datahub/omis/order/serializers.py]: Could make `ORDER_FIELDS_INVOICE_RELATED` a set if you don't need the order?

Also, the return value of dict.keys() is set-like so validated_data.keys() should work.

File: datahub/omis/order/test/factories.py [hunk 2]
```diff
@@ -92,18 +92,17 @@ class OrderWithCancelledQuoteFactory(OrderFactory):
     quote = factory.SubFactory(CancelledQuoteFactory)
 
 
+def _make_invoice(obj, *args, **kwargs):
+    obj.invoice = Invoice.objects.create_from_order(obj)
```
Comment (reupen) [inline datahub/omis/order/test/factories.py]: Should we check `create` here? I guess it's just the order reference that will be missing if create == False though.

File: datahub/omis/order/serializers.py (hunk 1 above)
Comment (marcofucci) [reply datahub/omis/order/serializers.py]: yes and yes, very good points, thanks!

File: datahub/omis/order/test/factories.py (hunk 2 above)
Comment (marcofucci) [reply datahub/omis/order/test/factories.py]: yep true, I've refactored it as well

**Pull request merged**